### PR TITLE
[GHA] ccache cleanup

### PR DESCRIPTION
**Pull request by mryzhov**
### Details:
 - Updated dependencies
 - Added ability to search cache files recursively
 - Enabled monthly automatic cleanup  

### Tickets:
 - *151079*

### CODE REVIEW

File: .github/workflows/cleanup_caches.yml [hunk 1]
```diff
@@ -1,14 +1,16 @@
 name: Cleanup caches
 on:
   workflow_dispatch:
+  pull_request:
```
Comment (akashchi) [inline .github/workflows/cleanup_caches.yml:4]: Should it be run on every PR? or is it for testing only?

File: .github/workflows/cleanup_caches.yml [hunk 2]
```diff
@@ -34,15 +36,16 @@ jobs:
...
+    name: Cleanup Linux ccache
     runs-on: aks-linux-2-cores-8gb
     if: ${{ github.repository_owner == 'openvinotoolkit' }}
     container:
       image: openvinogithubactions.azurecr.io/dockerhub/ubuntu:20.04
       volumes:
         - /mount:/mount
     env:
-      CCACHE_PATH: /mount/caches/ccache/ubuntu22_riscv64
+      CCACHE_PATH: /mount/caches/ccache
```
Comment (akladiev) [inline .github/workflows/cleanup_caches.yml:48]: @ababushk, it points to the Blob Storage, right? Isn't it supposed to cleanup old blobs by itself, or...?

File: .github/workflows/cleanup_caches.yml (hunk 1 above)
Comment (mryzhov) [reply .github/workflows/cleanup_caches.yml:4]: yes, that is for testing only

Comment (mryzhov) [inline .github/workflows/cleanup_caches.yml:4]:
```suggestion
```

**Pull request merged**